src/Adapters/Etw.Net451/ITraceEventSession.cs

```diff
@@ -0,0 +1,81 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ITraceEventSession.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.ApplicationInsights.EtwCollector
+{
+    using System;
+    using Diagnostics.Tracing.Session;
+    using Microsoft.Diagnostics.Tracing;
+
+    /// <summary>
+    /// Abstracts properties and methods of <see cref="Microsoft.Diagnostics.Tracing.Session.TraceEventSession"/> to provide better test ability.
+    /// </summary>
+    internal interface ITraceEventSession : IDisposable
+    {
+        /// <summary>
+        /// If this is a real time session you can fetch the source associated with the session
+        /// to start receiving events. Currently does not work on file based sources (we
+        /// expect you to wait until the file is complete).
+        /// </summary>
+        ETWTraceEventSource Source { get; }
+
+        /// <summary>
+        /// Is the current process Elevated (allowed to turn on a ETW provider). This is
```

Comment:
See the comment above. This is not technically true and in general, I do not think it makes sense to do this kind of checking in our code. Just rely on exceptions from TraceEvent library and/or the operating system